### PR TITLE
Create separate rosmon log for every node

### PR DESCRIPTION
**Pull request by lindawang-6**
https://www.wrike.com/open.htm?id=478608835

### CODE REVIEW

File: rosmon_core/src/main.cpp [hunk 1]
```diff
@@ -356,13 +357,13 @@ int main(int argc, char** argv)
...
+                        workDir = logDir = logDir + "/rosmon";
                         if (chdir(logDir.c_str()) == 0 || mkdir(logDir.c_str(), 0777) == 0) 
                         {
-                                std::string dir = logDir + "/roslogs";
-                                if (chdir(dir.c_str()) == 0 || mkdir(dir.c_str(), 0777) == 0) 
+                                logDir = logDir + "/roslogs";
+                                if (chdir(logDir.c_str()) == 0 || mkdir(logDir.c_str(), 0777) == 0) 
                                 {
-                                        logFile = dir + "/" + launchInfo.launch_group + "_" + launchInfo.launch_config + "_" + launchInfo.robot_name + ".log";
+					logFile = logDir + "/" + launchInfo.launch_group + "_" + launchInfo.launch_config + ".log";
```
Comment (lindawang-6) [inline rosmon_core/src/main.cpp:366]: Fixed spacing error from before.

File: rosmon_core/src/main.cpp [hunk 2]
```diff
@@ -491,7 +492,7 @@ int main(int argc, char** argv)
 
 	fmt::print("Running as '{}'\n", ros::this_node::getName());
 
-    rosmon::monitor::Monitor monitor(config, watcher);
+	rosmon::monitor::Monitor monitor(config, watcher, logDir, flushLog, launchInfo.launch_group, launchInfo.launch_config);
```
Comment (lindawang-6) [inline rosmon_core/src/main.cpp:495]: Fixed spacing error from before.

File: rosmon_core/src/main.cpp (hunk 1 above)
Comment (gauthamm) [inline rosmon_core/src/main.cpp:366]: we can drop this now, since you are writing to each node

File: rosmon_core/src/main.cpp [hunk 3]
```diff
@@ -340,6 +340,7 @@ int main(int argc, char** argv)
 
 	// Setup logging
 	boost::scoped_ptr<rosmon::Logger> logger;
+	std::string workDir;
 	{
 		// Setup a sane ROSCONSOLE_FORMAT if the user did not already
 		setenv("ROSCONSOLE_FORMAT", "[${function}]: ${message}", 0);
```
Comment (gauthamm) [inline rosmon_core/src/main.cpp]: make sure this has time too

File: rosmon_core/src/monitor/monitor.cpp [hunk 4]
```diff
@@ -32,16 +35,51 @@ namespace rosmon
...
+Monitor::Monitor(launch::LaunchConfig::ConstPtr config, FDWatcher::Ptr watcher, std::string logDir, bool flushLog, std::string launchGroup, std::string launchConfig)
  : m_config(std::move(config))
  , m_fdWatcher(std::move(watcher))
  , m_ok(true)
 {
 	for(auto& launchNode : m_config->nodes())
 	{
-		auto node = std::make_shared<NodeMonitor>(launchNode, m_fdWatcher, m_nh);
+		// Setup a sane ROSCONSOLE_FORMAT if the user did not already
+		setenv("ROSCONSOLE_FORMAT", "[${function}]: ${message}", 0);
```
Comment (gauthamm) [inline rosmon_core/src/monitor/monitor.cpp]: make sure this has time too

**Review by gauthamm — APPROVED** [review]

File: rosmon_core/src/main.cpp (hunk 1 above)
Comment (lindawang-6) [reply rosmon_core/src/main.cpp:366]: This log still provides information. If a node dies, it is written in this log.

**Review by gajena — APPROVED** [review]

**Pull request merged**